modules/phase_field/src/kernels/EpsilonModelKernel2ndV2Gauss.C

```diff
@@ -0,0 +1,31 @@
+#include "EpsilonModelKernel2ndV2Gauss.h"
+
+registerMooseObject("PhaseFieldApp", EpsilonModelKernel2ndV2Gauss);
+
+InputParameters
+EpsilonModelKernel2ndV2Gauss::validParams()
+{
+  InputParameters params = ADKernel::validParams();
+  params.addClassDescription("Implements the term: -L \\nabla \\cdot \\left( \\frac{\\partial "
+                             "m(\\theta, v)}{\\partial \\nabla \\eta_i} f_0 \\right)");
+
+  params.addParam<MaterialPropertyName>("f_name", "F", "Base name of the free energy function");
+
+  return params;
+}
+
+EpsilonModelKernel2ndV2Gauss::EpsilonModelKernel2ndV2Gauss(const InputParameters & parameters)
+  : ADKernel(parameters),
+    _L_AD(getADMaterialProperty<Real>("L_AD")),
+    _dm_minus(getADMaterialProperty<RealGradient>("dm_minus")),
+    _F(getMaterialProperty<Real>("F"))
```

Comment:
why not AD here?

Reply:
I was using TotalFreeEnergy auxkernel that required non-AD materials. But, I made a new AD version ADTotalFreeEnergy; so I do not have anymore non-AD materials